server/dive_server/utils.py

```diff
@@ -3,16 +3,38 @@
 import os
 from datetime import datetime
 from pathlib import Path
-from typing import Dict
+from typing import Dict, Iterable, Optional
 
+from girder.exceptions import RestException
 from girder.models.file import File
 from girder.models.folder import Folder
 from girder.models.item import Item
 from girder.models.upload import Upload
+from girder.models.user import User
 
 from dive_server.serializers import viame
 
 
+def all_detections_items(folder: Folder) -> Iterable[Item]:
+    """caller is responsible for verifying access permissions"""
+    return Item().find({"meta.detection": str(folder['_id'])}).sort([("created", -1)])
+
+
+def detections_item(folder: Folder, strict=False) -> Optional[Item]:
+    all_items = all_detections_items(folder)
+    detection = all_items[0] if all_items else None
+    if not detection and strict:
+        raise RestException(f"No detections for folder {folder['name']}")
+    return detection
+
+
+def detections_file(folder: Folder, strict=False):
+    item = detections_item(folder, strict)
+    if item is None and not strict:
+        return None
+    return Item().childFiles(item[0])[0]
```

Comment:
in `detections_item()` you already return `all_items[0]` so I think this might be an index into something that doesn't exist?